tests/unit/test_api_routing.py

```
@@ -0,0 +1,231 @@
+        self.assertEqual(response.status_code, 401)
+
+        data = response.get_json()
+        self.assertIn('error', data)
+        self.assertIn('Unauthorized', data['error'])
+
+    def test_predict_endpoint_with_auth(self):
+        """Test predict endpoint works with valid API key."""
+        response = self.app.post('/api/predict',
+                                 data=json.dumps({'text': 'I am happy'}),
+                                 content_type='application/json',
+                                 headers={'X-API-Key': cls.ADMIN_KEY})
```

Ruff (0.12.2)

Undefined name `cls`

(F821)